joomla/joomla-cms
# [com_modules] modules view (modal) - Implement searchtools

#### Description

This PR adds searchtools to com_modules modal layout modules view.

Also, in normal layout:
- improves responsive visual (mobile/tablet/desktop)
- the acess column now orders by access level designation and not by access level id (it doesn't make a lot of sense order by the access level ID ... but makes sense order by the acess level DESIGNATION).
- code improvements
###### Before PR (modal)

![before-pr](https://cloud.githubusercontent.com/assets/9630530/13374056/3c48b116-dd71-11e5-90e4-273dda043db9.png)
###### After PR  (modal)

![after-pr](https://cloud.githubusercontent.com/assets/9630530/13374104/0c9bd3a6-dd73-11e5-9ff2-e3a41870e507.png)
#### How to test

Normal view:
1. Install latest staging and apply this patch
2. Go to Extensions -> Modules
3. Check if search, filters, ordering and pagination are working properly
4. Test on mobile and tablet (ex: reduce browser width)

Modal view (test in frontend and backend):
1. Go to any content and click on "+Module" button on the editor to open the modal layout
2. Check if search, filters, ordering and pagination are working properly.
3. Test selecting a position or a module.
#### Observations
1. Changed the modal default list when in the backend, now also shows other states (trashed, archved and unpublished). Why? To show the same states as "+Article" modal - consistency. In the frontend only shows enabled modules in the current language or in All language.
2. Added a new `/components/com_modules/models/forms/filter_modules.xml` file to control the frontend searchtools fields (a similiar one already exists for the com_content in `/components/com_content/models/forms/filter_articles.xml`). Please tell if there are any problems with this.
